src/test/playermanagertest.cpp

```diff
@@ -186,6 +193,46 @@ TEST_F(PlayerManagerTest, UnEjectInvalidTrackIdTest) {
     m_pPlayerManager->slotSaveEjectedTrack(pTrack);
     auto deck1 = m_pPlayerManager->getDeck(1);
     // Does nothing -- no crash.
+    // make sure eject does not trigger 'unreplace'
+    QTest::qSleep(500);
     deck1->slotEjectTrack(1.0);
     ASSERT_EQ(nullptr, deck1->getLoadedTrack());
 }
+
+TEST_F(PlayerManagerTest, UnReplaceTest) {
+    // Trigger eject twice within 500 ms to undo track replacement
+    auto deck1 = m_pPlayerManager->getDeck(1);
+    // Load a track
+    TrackPointer pTrack1 = getOrAddTrackByLocation(getTestDir().filePath(kTrackLocationTest1));
+    ASSERT_NE(nullptr, pTrack1);
+    TrackId testId1 = pTrack1->getId();
+    ASSERT_TRUE(testId1.isValid());
+    deck1->slotLoadTrack(pTrack1, false);
+    m_pEngine->process(1024);
+    while (!deck1->getEngineDeck()->getEngineBuffer()->isTrackLoaded()) {
+        QTest::qSleep(100); // millis
+    }
+    ASSERT_NE(nullptr, deck1->getLoadedTrack());
+
+    // Load another track.
+    TrackPointer pTrack2 = getOrAddTrackByLocation(getTestDir().filePath(kTrackLocationTest2));
+    ASSERT_NE(nullptr, pTrack2);
+    deck1->slotLoadTrack(pTrack2, false);
+    m_pEngine->process(1024);
+    while (!deck1->getEngineDeck()->getEngineBuffer()->isTrackLoaded()) {
+        QTest::qSleep(100); // millis
+    }
+    ASSERT_NE(nullptr, deck1->getLoadedTrack());
+
+    // Eject. Make sure eject does not trigger 'unreplace':
+    // sleep for longer than 500 ms 'unreplace' period so this is not registered as double-click
+    QTest::qSleep(500); // millis
```

Comment:
can we make the duration a constant in `basetrackplayer.h` and reuse it instead of hardcoding it everywhere?

Reply:
Jup, done.